### PR TITLE
Some fixes and additions

### PR DESCRIPTION
**Pull request by the-j0k3r**
@BrainSlayer please review

if this is OK use this patch link to apply upstream https://patch-diff.githubusercontent.com/raw/mirror/dd-wrt/pull/49.patch

### CODE REVIEW

**Comment by BrainSlayer** [comment]
looks okay

**Comment by the-j0k3r** [comment]
/me waits for next build including all this and other stuff at edge of seat

famous last words

We need to discuss the dd-wrt favicon(s) at some stage, unless you're very attached to these old designs I would propose forum/trac/fw all have their unique ones because trying to wack dd-wrt + wifi bands in a 16x16 pixel is going to be unreadable unrecognizable on a browser tab.

I looked at the wiki old logo with the aerials + fist but not sure if I like that fist idea, cof reminds me of too political themes.

**Comment by BrainSlayer** [comment]
i can ready the favicon without issues in my browser tab. for trac etc. yeah makes sense to change the default favicon to the same one. if you have a better idea for a new favicon, just make a proposal. but the current one works and is readable

**Comment by the-j0k3r** [comment]
> but the current one works and is readable

No one can disagree with that because `something` shows up! Readable idk if that's my honest experience.
Capture

its just too much going on, at pixel level it looks like this

Capture

here is some context for your eyes that is the firmware favicon in between those other favicons.

Capture

So Im interested in what you see that Im missing =) My interest is to make relevant in a not IE browser world.

**Comment by BrainSlayer** [comment]
> > but the current one works and is readable
> 
> No one can disagree with that because `something` shows up! Readable idk if that's my honest experience. Capture
> 
> its just too much going on, at pixel level it looks like this
> 
> Capture
> 
> here is some context for your eyes that is the firmware favicon in between those other favicons.
> 
> Capture
> 
> So Im interested in what you see that Im missing =) My interest is to make relevant in a not IE browser world.

yeah. you should maybe display the favicon with its correct colors. this favicon you show here is not what it should look like. inverting colors might confuse you, but this is a user eror

**Comment by BrainSlayer** [comment]
image

**Comment by the-j0k3r** [comment]
On white?

Capture

But I guess then its not worth me wasting time then. Not interested in white anything.

**Comment by BrainSlayer** [comment]
standard desktop color. you cannot do a favicon with transparent background if the user decides to use non standard colors. there is no way for it it will always collide withe foreground and it will always look wrong. you can now do a icon with black background. it will look wrong then on most other desktops

**Comment by BrainSlayer** [comment]
and a favicon isnt limited to 16x16. 32x32 and 48x48 is also possible or even multiple icons

**Comment by the-j0k3r** [comment]
Im aware what the limitations are, its possible with specific designs to work on both backgrounds, as favicons go on standard browsers 16x16 is the defining core design for anything bigger than. Its my experience for best designs that work regardless.

Its not possible with the current dd-wrt favicon design to work on both, since it only accounts for the default and the default is true for such a significant number that the rest use cases are simply ignored by most designers even today.

Its possible with right design and color choices to elicit contrast no matter the choices of background. But even modern designers and brand builders fail, because they choose only one or the other and never both, when they sacrifice nothing with a broader and simpler design choices,

TL;DR

Well, thats all she wrote. the victim died with 4 bullets in the eye sockets.

**Comment by BrainSlayer** [comment]
the problem is the background can be "any" color. not just black and white. so there is no solution than using different favicon icons depending on the known themes 
https://stackoverflow.com/questions/55170708/detect-if-the-browser-is-using-dark-mode-and-use-a-different-favicon

**Comment by the-j0k3r** [comment]
That's a programmatic solution to a design oversight in any case. From inception there doesn't have to be an issue to workaround.

A fair example to quote is openwrt's favicon, however dd-wrt doesn't have a simple iconographic logo that translates well into any size so its a moot point. To address this thats takes a great deal of time to reach a consensus on design choices and iterations which messes with recognizable branding and rebranding isn't cheap.

I think the simplest solution is always better and cheaper and I'll just override the issues locally with a design that didn't meet the cut I had done a few years back for a customer icon set. After all, I'm the 0.1333333% of users and dd-wrt has other issues that could be better make use of your time and particular expertise on that front.

Thanks for your time and attention to this though. I'll have to think about this for a bit.

In the meanwhile https://www.logo.bot spat out a router. These sites will do me out of a job I swear. lol

**Comment by BrainSlayer** [comment]
nice site. but the results are tried are all looking like from a japanese manga

**Comment by the-j0k3r** [comment]
> nice site. but the results are tried are all looking like from a japanese manga

All bells and no whistles right? All the graphics are what I call pre-ejaculated available for a dollar on every corner of the webs.
